Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Object Bucket Claim CRD
+weight: 2800
+indent: true
+---
+
+# Ceph Object Bucket Claim
+
+Rook supports the creation of new buckets (and access to existing buckets) via two custom resources:
+
+- an `Object Bucket Claim (OBC)` is custom resource which requests a bucket (new or existing) and is described by a Custom Resource Definition (CRD) shown below.
+- an `Object Bucket (OB)` is a custom resource generated when a bucket is provisioned and is described by an OB CRD, also shown below.
+
+An OBC references a storage class which is created by an administrator. The storage class defines whether the bucket requested is a new bucket or an existing bucket. It also defined the bucket retention policy.
+
+## Sample
+##### Storage class
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-bucket
+provisioner: ceph.rook.io/bucket
+reclaimPolicy: Delete
+parameters:
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-east-1
+  #bucketName:
+```
+##### Object Bucket Claim
+```yamlapiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-bucket
+spec:
+  #bucketName:
+  generateBucketName: ceph-bkt
+  storageClassName: rook-ceph-bucket
+```
+
+## Object Bucket Claim Settings
+
+### Storage Class
+
+##### Metadata
+
+- `name`: The name of the storage class need to create, which will be reflected in the object bucket claim.
+- `reclaimpolicy`: two policies supported `Delete` and `Retain`, see explaination as [specified in Kubernetes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain)
+
+##### Parameters
+- `objectStoreName` : The name of the object store.
+- `objectStoreNamespace` : The name space where object store resides.
+- `region`: The region where object store resides.
+- `bucketName`: For brownfield cases, a pre-existing bucket that a user is granted access to. It will not create a new bucket.
```

Comment:
Please change to:
" `bucketName`: Required for brownfield cases. Names a pre-existing bucket that a user will be granted access to. A new bucket will not be created." Must be omitted for greenfield cases since the OBC will either name a specific bucket or will ask the system to generate a random bucket name."